gobblin-utility/src/main/java/org/apache/gobblin/util/JobLauncherUtils.java

```diff
@@ -132,10 +132,9 @@ public static void cleanStagingData(List<? extends State> states, Logger logger)
    * @param logger a {@link Logger} used for logging
    */
   public static void cleanJobStagingData(State state, Logger logger) throws IOException {
-    Preconditions.checkArgument(state.contains(ConfigurationKeys.WRITER_STAGING_DIR),
-        "Missing required property " + ConfigurationKeys.WRITER_STAGING_DIR);
-    Preconditions.checkArgument(state.contains(ConfigurationKeys.WRITER_OUTPUT_DIR),
-        "Missing required property " + ConfigurationKeys.WRITER_OUTPUT_DIR);
+    if (!state.contains(ConfigurationKeys.WRITER_STAGING_DIR) || !state.contains(ConfigurationKeys.WRITER_OUTPUT_DIR)) {
```

Comment:
Shouldn't this be invoked after the workunit stream has been computed? If so, shouldn't we expect both these config keys to be set?

Reply:
This looks like a change in behavior from what existed earlier.

Reply:
This checks the job level state, not the workunit. So it would fail if we do not specify the temporary directories.